src/zenml/integrations/integration.py

```diff
@@ -43,6 +43,17 @@ def __new__(
         cls = cast(Type["Integration"], super().__new__(mcs, name, bases, dct))
         if name != "Integration":
             integration_registry.register_integration(cls.NAME, cls)
+
+        if getattr(cls, "REQUIREMENTS"):
+            logger.warning(
+                f"Integration {cls.NAME} has a REQUIREMENTS attribute. "
+                f"Please use the get_requirements() method instead."
+                f"REQUIREMENTS will be deprecated in the future."
+            )
+
+        get_requirements = getattr(cls, "get_requirements", None)
+        if get_requirements and callable(get_requirements):
+            cls.REQUIREMENTS = cls.get_requirements()
```

Comment:
The entire target os only makes sense if we then do not statically set something once. Instead:
- `get_requirements(...) by default returns `cls.REQUIREMENTS`
- In all places where we use `SomeIntegration.REQUIREMENTS` you should now use `SomeIntegration.get_requirements(...)` and optionally specify a target os if necessary (this is for example for the docker builds)

Reply:
@schustmi should the target_os be defaulted into specific target_os, `Linux` for example?

Reply:
I think the default should be your local OS (e.g. if you want to do `zenml integration install ...` you want the packages for your local os. Only docker builds need to explictly request packages for linux